### DISCUSSION TITLE
How to add session features during ARCore initialization in ARSceneView?

### DISCUSSION THREAD

**Discussion by Kliuchko** [sceneview/sceneview#457] [Q&A]
Hi guys! Am I missing something basic or is there really a bug in the project? How do I add a Session Feature during ARCore initialization if in the ARSceneView class the "sessionFeatures" field looks like this: val sessionFeatures: Set<Session.Feature> = setOf(). Do I understand correctly that there is no way to add a new element to the set or am I missing something very simple?

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, use the `sessionConfiguration` parameter:

```kotlin
ARSceneView(
    sessionConfiguration = { session, config ->
        // Configure AR features here
        config.planeFindingMode = Config.PlaneFindingMode.HORIZONTAL_AND_VERTICAL
        config.depthMode = Config.DepthMode.AUTOMATIC
        config.lightEstimationMode = Config.LightEstimationMode.ENVIRONMENTAL_HDR
        config.geospatialMode = Config.GeospatialMode.ENABLED
        // etc.
    }
) {
    // Your AR content
}
```

The `sessionConfiguration` callback gives you direct access to the ARCore `Session` and `Config` objects, so you can enable any ARCore feature.
